src/librustdoc/html/render/mod.rs

```diff
@@ -113,6 +113,10 @@ crate struct Context<'tcx> {
     /// real location of an item. This is used to allow external links to
     /// publicly reused items to redirect to the right location.
     crate render_redirect_pages: bool,
+    /// `None` by default, depends on the `generate-redirect-map` option flag. If this field is set
+    /// to `Some(...)`, it'll store redirections and then generate a JSON file at the top level of
+    /// the crate.
+    crate redirections: Option<Rc<RefCell<FxHashMap<String, String>>>>,
```

Comment:
Why do you need the `Rc`?

Reply:
Force of habit. I'll check if it's really necessary.

Reply:
No I didn't yet. Rebasing and then let's check!

Reply:
I can confirm it's required, without it there is no redirect file generated, meaning that it's not fulfilled as one would expect. I think it's because we need to use the same content through the various clones.